Engine/EditorUI/Panel/InspectorSubpanel/PanelComponent.cpp

```diff
@@ -164,55 +164,33 @@ void PanelComponent::ShowComponentParticleSystem(ComponentParticleSystem* partic
 			App->actions->DeleteComponentUndo(particle_system);
 		}
 		ImGui::Separator();
-
-		if (ImGui::CollapsingHeader("Particle Values", ImGuiTreeNodeFlags_DefaultOpen))
+		if (ImGui::CollapsingHeader("Debug", ImGuiTreeNodeFlags_DefaultOpen))
 		{
-			ImGui::Text("Texture");
-			ImGui::SameLine();
-
-			std::string texture_name = particle_system->billboard->billboard_texture == nullptr ? "None (Texture)" : App->resources->resource_DB->GetEntry(particle_system->billboard->billboard_texture->GetUUID())->resource_name;
-			ImGuiID element_id = ImGui::GetID((std::to_string(particle_system->UUID) + "TextureSelector").c_str());
-			if (ImGui::Button(texture_name.c_str()))
+			static int emit_count = 0;
+			ImGui::InputInt("##Emit_amount", &emit_count);
+			if (ImGui::Button("Emit"))
 			{
-				App->editor->popups->resource_selector_popup.ShowPanel(element_id, ResourceType::TEXTURE);
+				particle_system->Emit(emit_count);
 			}
-
-			uint32_t selected_resource_uuid = App->editor->popups->resource_selector_popup.GetSelectedResource(element_id);
-			if (selected_resource_uuid != 0)
+			if (ImGui::Button("Start"))
 			{
-				particle_system->SetParticleTexture(selected_resource_uuid);
-				particle_system->modified_by_user = true;
+				particle_system->Play();
 			}
-			selected_resource_uuid = ImGui::ResourceDropper<Texture>();
-			if (selected_resource_uuid != 0)
+			ImGui::SameLine();
+			if (ImGui::Button("Stop"))
 			{
-				particle_system->SetParticleTexture(selected_resource_uuid);
-				particle_system->modified_by_user = true;
+				particle_system->Stop();
```

Comment:
same as before, the UI is ok, but take in mind that there should be a pause button aswell ^^